docs/contributing-to-airbyte/tutorials/building-a-java-destination.md

```diff
@@ -0,0 +1,174 @@
+# Building a Java Destination
+
+## Summary
+
+This article provides a checklist for how to create a Java destination. Each step in the checklist has a link to a more detailed explanation below.
+
+## Requirements
+
+Docker and Java with the versions listed in the [tech stack section](../../understanding-airbyte/tech-stack.md).
+
+## Checklist
+
+### Creating a destination
+
+* Step 1: Create the destination using the template generator
+* Step 2: Build the newly generated destination 
+* Step 3: Implement `spec` to define the configuration required to run the connector
+* Step 4: Implement `check` to provide a way to validate configurations provided to the connector
+* Step 5: Implement `write` to write data to the destination
+* Step 6: Set up Acceptance Tests
+* Step 7: Write unit tests or integration tests
+* Step 8: Update the docs \(in `docs/integrations/destinations/<destination-name>.md`\)
+
+{% hint style="info" %}
+All `./gradlew` commands must be run from the root of the airbyte project.
+{% endhint %}
+
+{% hint style="info" %}
+If you need help with any step of the process, feel free to submit a PR with your progress and any questions you have, or ask us on [slack](https://slack.airbyte.io).
+{% endhint %}
+
+## Explaining Each Step
+
+### Step 1: Create the destination using the template
+
+Airbyte provides a code generator which bootstraps the scaffolding for our connector.
+
+```bash
+$ cd airbyte-integrations/connector-templates/generator # assumes you are starting from the root of the Airbyte project.
+$ ./generate.sh
+```
+
+Select the `Java Destination` template and then input the name of your connector. We'll refer to the destination as `<name>-destination` in this tutorial, but you should replace `<name>` with the actual name you used for your connector e.g: `BigQueryDestination` or `bigquery-destination`. 
+
+### Step 2: Build the newly generated destination
+
+You can build the destination by running:
+
+```bash
+# Must be run from the Airbyte project root
+./gradlew :airbyte-integrations:connectors:destination-<name>:build
+```
+
+this compiles the java code for your destination and builds a Docker image with the connector. At this point, we haven't implemented anything of value yet, but once we do, you'll use this command to compile your code and Docker image. 
+
+{% hint style="info" %}
+    Airbyte uses Gradle to manage Java dependencies. To add dependencies for your connector, manage them in the `build.gradle` file inside your connector's directory.   
+{% endhint %}
+
+#### Iterating on your implementation
+
+We recommend the following ways of iterating on your connector as you're making changes:
+
+* Test-driven development (TDD) in Java 
+* Test-driven development (TDD) using Airbyte's Acceptance Tests
+* Directly running the docker image 
+
+**Test-driven development in Java**
```

Comment:
I feel like you may as well make these H4 instead of just bolding them since you did that with the previous subheading. Same for the following bolded subheaders.
```suggestion
#### Test-driven development in Java
```